### DISCUSSION TITLE
Automatically trigger a curation peset

### DISCUSSION THREAD

**Discussion by PepijnVDK1** [awcodes/filament-curator#468] [Q&A]
Is there a way to automatically trigger a curation preset?
This would come in handy to automatically generate thumbnails and other preset for optimization.

Kind regards

**Comment by benjaminniess** [comment]
I might have the same question. What I'd like to do is programmatically generate as many images as I have in the preset configuration like WordPress does with [add_image_size](https://developer.wordpress.org/reference/functions/add_image_size/).

Thank you for your work by the way 🙏

**Reply by awcodes** [reply]
I guess you could extend the Media model or observer to create them using the observer but that’s not really the way the plugin is meant to be used. Curation’s are meant to be one off focal point corrections for images at certain sizes. Other wise you should use the glider blade component to render the images at the sizes you need. It generates them on the fly and caches them. This keeps you from having wasted storage on your server for a bunch of image size that might not ever get served.

**Reply by benjaminniess** [reply]
Thank you for your answer. The only thing with the glider component is that I doesn't seem possible to pre-render cached images so it can affect performances. I'll give it a go anyway. Thanks again.

**Reply by awcodes** [reply]
The first request for the image is the only performance hit, after that all images are served from the cache. It's no different than how Laravel caches views.

**Reply by benjaminniess** [reply]
Fair enough 👍

**Reply by JVariance** [reply]
But those Glide-URLs with parameters aren't that great for SEO, right? @awcodes

**Reply by JVariance** [reply]
I think we also had a 50x-Error from an outdated (cache duration exceeded) link output in a SEO-Tool
